src/main/java/org/victorrobotics/frc/dtlib/log/DTLogger.java

```diff
@@ -0,0 +1,202 @@
+  private static final String LOG_DIRECTORY      = "/dev/sda/logs";
+
+  static final Map<Class<?>, DTLogType<?>> LOGGABLE_TYPES = new HashMap<>();
+
+  static {
+    DTLogBuiltinTypes.load();
+  }
+
+  private static File        file;
+  private static DTLogWriter dataWriter;
+
+  private static long lastTimestamp;
+  private static long startTimeMillis;
+  private static int  nextHandle = 0x0100;
+
+  private DTLogger() {}
+
+  public static DTLogWriter getWriter() {
+    return dataWriter;
+  }
+
+  public static int newHandle(int typeID, String path) {
+    dataWriter.writeShort(typeID);
+    dataWriter.writeStringUTF8(path);
+    return nextHandle++;
+  }
+
+  public static boolean logNewTimestamp() {
+    long newTime = RobotController.getFPGATime() / 1000;
+    long diff = newTime - lastTimestamp;
+    if (diff <= 0) {
+      // Same time, no change
+      return false;
+    }
+    lastTimestamp = newTime;
+
+    if (diff <= 0xFFFF) {
+      // 2 byte form, maximum of 65.535 seconds
+      dataWriter.writeShort(0x0001);
+      dataWriter.writeShort((short) diff);
+    } else {
+      // write new timestamp
+      dataWriter.writeShort(0x0002);
+      dataWriter.writeLong(newTime);
+    }
+    return true;
+  }
+
+  public static boolean init() {
+    if (!isTimeSynchronized()) {
+      return false;
+    }
+
+    try {
+      openCapture();
+      writeHeader();
+      return true;
+    } catch (IOException e) {
+      try {
+        if (dataWriter != null) {
+          dataWriter.close();
+        }
+      } catch (IOException e2) {
+      }
+      dataWriter = null;
+      return false;
+    }
+  }
+
+  private static void openCapture() throws IOException {
+    Instant now = Clock.systemUTC()
+                       .instant();
+    lastTimestamp = RobotController.getFPGATime() / 1_000;
+    startTimeMillis = now.toEpochMilli();
+
+    String name = TIME_FORMATTER.format(now) + ".dtlog";
+    file = new File(LOG_DIRECTORY, name);
+    dataWriter = new DTLogWriter(file, 65536);
+  }
+
+  private static void writeHeader() {
+    dataWriter.writeBytes(DTLIB_LOGGER_MAGIC.getBytes(StandardCharsets.UTF_8));
+
+    dataWriter.writeShort(DTLibInfo.Version.YEAR);
+    dataWriter.writeByte(DTLibInfo.Version.MAJOR);
+    dataWriter.writeByte(DTLibInfo.Version.MINOR);
+
+    String[] wpilibVersion = WPILibVersion.Version.split("\\.");
+    dataWriter.writeShort(Integer.parseInt(wpilibVersion[0]));
+    dataWriter.writeByte(Integer.parseInt(wpilibVersion[1]));
+    dataWriter.writeByte(Integer.parseInt(wpilibVersion[2]));
+
+    dataWriter.writeShort(getTeamNumber());
+    dataWriter.writeShort(0); // unused flags
+    dataWriter.writeLong(startTimeMillis);
+  }
+
+  private static boolean isTimeSynchronized() {
+    return DriverStation.isDSAttached() && LocalDate.now(Clock.systemUTC())
+                                                    .getYear() >= 2000;
+  }
+
+  public static void logDebug(String msg) {
+    logMessage(msg, 0x0008);
```

Comment:
There seem to be several sets of numeric enumerated constants. It's not clear which of them form sets (and therefore require uniqueness) and which ones can coincidentally have the same values, because they aren't used in the same context. Declaring these constants together (in blocks/nested classes) would help clarify.